parsers/FR_O.py

```diff
@@ -1,351 +1,188 @@
-        datetime = arrow.now(tz=params["timezone"])
-    else:
-        datetime = arrow.get(target_datetime, params["timezone"])
+STORAGE_MAPPING = {
+    API_TYPE: type
+    for key in ["storage"]
+    for type, groups in API_PARAMETER_GROUPS[key].items()
+    for API_TYPE in groups
+}
 
-    # Data retrievement
-    response = r.get(api, params=params)
-    data = json.loads(response.content)
-    datapoints = []
 
-    # Response build
-    if len(data["records"]) > 0 and ("fields" in data["records"][0]):
-        for i in range(0, len(data["records"])):
-            fields = data["records"][i]["fields"]
-            datetime_result = arrow.get(fields[date_name]).datetime
-            result = {
-                "zoneKey": zone_key,
-                "datetime": datetime_result,
-                "source": get_source(zone_key, target_datetime),
-                "production": {
-                    "nuclear": 0,
-                    "biomass": 0,
-                    "coal": 0,
-                    "gas": 0,
-                    "hydro": 0,
-                    "oil": 0,
-                    "solar": 0,
-                    "wind": 0,
-                    "geothermal": 0,
-                    "unknown": 0,
-                },
-            }
-            # Non-thermal sources
-            for source_mapping_key in sources_mapping:
-                current_sources = sources_mapping[source_mapping_key]
-                for current_source in current_sources:
-                    if current_source in fields:
-                        value = fields[current_source]
-                        if value > 0:
-                            result["production"][source_mapping_key] += value
-
-            # Thermal sources
-            for k in range(0, len(thermal_mapping[zone_key])):
-                current_thermal = thermal_mapping[zone_key][k]
-                current_type = current_thermal["type"]
-                current_source = None
-                if current_thermal["monthly_variation"]:
-                    current_source = current_type[datetime.month]
-                else:
-                    current_source = current_type
-                for type_name in current_source:
-                    if current_thermal["name"] in fields:
-                        value = fields[current_thermal["name"]]
-                        multiple = current_source[type_name]
-                        result["production"][type_name] += value * multiple
+def generate_url(zone_key, target_datetime):
+    return f"{DOMAIN_MAPPING[zone_key]}/api/v2/catalog/datasets/{HISTORICAL_DATASETS[zone_key] if target_datetime else LIVE_DATASETS[zone_key]}/exports/json"
 
-            datapoints.append(result)
 
-    return datapoints
+def fetch_data(
+    zone_key: str = "FR-COR",
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger=getLogger(__name__),
+) -> Tuple[Any, str]:
+    ses = session or Session()
```

Comment:
It would be more clear to have Session as a default value for the function.
So:
def fetch_data(zone_key:str, session = Session(),....)

Reply:
I know that other languages will require a compile constant but hey, let's take advantage that Python is interpreted :)

Reply:
Agreed but this is the way we do it in the other parsers so I felt consistensy is more important than "correctness".

I'd be happy to change it though.

Reply:
I think we could update that at a later stage, it would simplify stuff.